synapse/storage/databases/main/events_worker.py

```diff
@@ -602,7 +602,7 @@ async def _get_events_from_cache_or_db(
             # already due to `_get_events_from_db`).
             fetching_deferred: ObservableDeferred[
                 Dict[str, _EventCacheEntry]
-            ] = ObservableDeferred(defer.Deferred())
+            ] = ObservableDeferred(defer.Deferred(), consumeErrors=True)
```

Comment:
Without `consumeErrors`, twisted thinks that exceptions don't get handled.

Reply:
one has to wonder if there is ever a usecase for ObservableDeferred to *not* `consumeErrors`.

This inspired me to write https://github.com/matrix-org/synapse/issues/11390.